mindsdb/integrations/handlers/llama_index_handler/llama_index_handler.py

```diff
@@ -0,0 +1,108 @@
+import os
+from typing import Optional, Dict
+
+from llama_index import GPTVectorStoreIndex,download_loader
+import pandas as pd
+
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+from mindsdb.utilities.log import get_log
+
+
+logger = get_log("integrations.llama_index_handler")
+
+
+
+class LlamaIndexHandler(BaseMLEngine):
+    """
+    Integration with the LlamaIndex Python Library
+    """
+    name = 'llama_index'
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.default_index_class  = 'GPTVectorStoreIndex'
+        self.default_query_engine = 'as_query_engine' 
+        self.supported_index_class = ['GPTVectorStoreIndex']
+        self.supported_query_engine = ['as_query_engine'] 
+
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+
+        if 'using' not in args:
+            raise Exception("LlamaIndex engine requires a USING clause! Refer to its documentation for more details.")
+
+        if 'source_url_link' not in args['using']:
+            raise Exception("LlamaIndex engine requires a source_url_link parameter.Refer to its documentation for more details.")
+
+        if 'openai_api_key' not in args['using']:
+            raise Exception("LlamaIndex engine requires a openai_api_key parameter.Refer to its documentation for more details.")
+
+        if 'index_class' not in args['using']:
+            args['using']['index_class'] = self.default_index_class
+        elif args['using']['index_class'] not in self.supported_index_class:
+            raise Exception(f"Invalid index class argument. Please use one of {self.supported_index_class}")
+
+        if 'query_engine' not in args['using']:
+            args['using']['query_engine'] = self.default_query_engine
+        elif args['using']['query_engine'] not in self.supported_query_engine:
+            raise Exception(f"Invalid operation mode. Please use one of {self.supported_query_engine}")
+
+        self.model_storage.json_set('args', args)
+
+
+    def predict(self, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        
+        args = self.model_storage.json_get('args')
+
+        input_keys = list(args.keys())
+
+        os.environ['OPENAI_API_KEY'] = args['using']['openai_api_key']
+
+        SimpleWebPageReader = download_loader("SimpleWebPageReader")
+        documents = SimpleWebPageReader(html_to_text=True).load_data([args['using']['source_url_link']])
+
+        index = GPTVectorStoreIndex.from_documents(documents)
+    
+        query_engine = index.as_query_engine()
+        response = query_engine.query(df['question'].iat[0])
```

Comment:
Would this support bulk predictions? 

If a MindsDB user does this:
```sql
SELECT t.question, m.answer
FROM mindsdb.my_qa_model as m
JOIN my_data.table_with_a_bunch_of_questions as t;
```

they expect as output a table with the model's answer for every question in the second clause of the join. Based on the `iat[0]`, I would imagine it will not work (maybe I'm missing something though!). 

Please double check that the above example does go through, as it should definitely be part of the initial release.

Reply:
Thanks, @paxcema for reviewing the PR and pointing out this.
 I fixed this and updated the README.md file with SQL usage. Please find the attached working screenshots of the same.